packages/now-routing-utils/src/superstatic.ts

```diff
@@ -110,8 +110,9 @@ function isString(key: any): key is string {
 function replaceSegments(segments: string[], destination: string): string {
   if (destination.includes(':')) {
     segments.forEach((name, index) => {
-      const r = new RegExp(':' + name, 'g');
-      destination = destination.replace(r, toSegmentDest(index));
+      const part = `:${name}`;
+      const r = new RegExp(`(${part}\\*|${part}\\+|${part})`, 'g');
```

Comment:
I don't think this works as expected.

For example, if `name = 'user'` then this regex will match the string `"users"`.

Reply:
How about this:

```js
const segment = name.endsWith('*') || name.endsWidth('+') ? name.slice(0, -1) : name;
const r = new RegExp(':' + segment, 'g');
```

Reply:
The segments don't include the `*` or `+` character, the above regex shouldn't match `users` as `*` and `+` are being escaped

Reply:
```js
var name = 'user';
var part = `:${name}`;
var r = new RegExp(`(${part}\\*|${part}\\+|${part})`, 'g');
var destination = '/:user/edit';
console.log(destination.replace(r, 'jj')); // matched, good

destination = '/:users/edit';
console.log(destination.replace(r, 'jj')); // matched, bad
```